lib/net.c

```diff
@@ -88,11 +96,18 @@ gatekeeper_init_network(struct net_config *net_conf)
 		}
 	}
 
-	num_lcores = rte_lcore_count();
-	
 	RTE_ASSERT(net_conf->num_rx_queues <= GATEKEEPER_MAX_QUEUES);
 	RTE_ASSERT(net_conf->num_tx_queues <= GATEKEEPER_MAX_QUEUES);
 
+	/* The master lcore doesn't listen on a queue. */
+	RTE_ASSERT(net_conf->num_rx_queues < rte_lcore_count());
+	RTE_ASSERT(net_conf->num_tx_queues < rte_lcore_count());
+
+	/* Assign queue identifier to each lcore. */
+	RTE_LCORE_FOREACH_SLAVE(lcore) {
```

Comment:
The queue arrays should be initialized with an invalid queue identifier. When `lcore_to_queue()` sees an invalid number, it allocates a queue identifier, and returns that. The following calling will return the previously allocated identifier.